### PR TITLE
Add explicit date

### PR DESCRIPTION
**Pull request by ericholscher**

### CODE REVIEW

File: migrate-configuration-v2.rst
Comment (benjaoming) [reply migrate-configuration-v2.rst:18]: A quick note: We can use "brownout" internally, but it's not gonna make much sense in a public announcement :)

Comment (agjohnson) [reply migrate-configuration-v2.rst:18]: Brownout is a term fairly commonly used here, but probably worth using some supporting language here as well. Here is a GitHub announcement for comparison:
https://github.blog/changelog/2021-08-10-brownout-notice-api-authentication-via-query-parameters-for-48-hours/

Also noting, this was the third brownout and it lasted 48h. I wonder if instead of 2h blocks we instead just do something like a 12h period (July 24) and a 24h period (Aug 7).

Comment (benjaoming) [reply migrate-configuration-v2.rst:18]: I see! allow me to revise that: We cannot use brownout without accompanying it with a good description of what it means :)

**Review by benjaoming — APPROVED** [review]
In order to have the date visible quickly, I'm fine about merging this and adding other dates in a second iteration.

But if someone adds the brownout dates (+ nice with a definition of what we mean by that) in this PR, that's great, too :+1:

File: migrate-configuration-v2.rst
Comment (agjohnson) [inline migrate-configuration-v2.rst:26]: "It" is in wrong agreement here ("defaults")

```suggestion
we found them impossible to change over time.
```

Comment (agjohnson) [reply migrate-configuration-v2.rst:26]: Or

```suggestion
we found it impossible to change them over time.
```

File: migrate-configuration-v2.rst
Comment (agjohnson) [inline migrate-configuration-v2.rst:33]:
```suggestion
We understand this change will affect many of our users,
```

File: migrate-configuration-v2.rst
Comment (agjohnson) [inline migrate-configuration-v2.rst:38]: Brought up in comments here, but maybe lets make this even easier on us and replicate what GitHub did: 12h window first date, 24h window the second date. I'm guessing it might be hard to have folks around for all of these 2h windows, thinking about this a bit more. We'll definitely catch more projects this way too.

File: migrate-configuration-v2.rst
Comment (benjaoming) [reply migrate-configuration-v2.rst:10]: Yes, the filename is in the headline right above, so that's fine :+1:

File: migrate-configuration-v2.rst
Comment (agjohnson) [reply migrate-configuration-v2.rst:38]: If we start a 12h window Mon Sept 4 @ 0:00 PT and go till 12:00 PT, this is what the overlap looks like:

image

Asia users will get errors starting at the end of their day and the errors will resolve overnight, so not *great*, but we have more users in EU/US anyways.

Comment (ericholscher) [reply migrate-configuration-v2.rst:38]: Sounds good to me 👍

File: migrate-configuration-v2.rst
Comment (ericholscher) [reply migrate-configuration-v2.rst:33]: I always mess this one up 🙃

**Comment by agjohnson** [comment]
Note, before merging this, urllib3 might not be planning to add a deprecation exception like we saw in urllib3 2.0. This would allow us to relax the timeframe a little bit, and maybe aim for end of Sept instead. Still waiting on confirmation here, in #222

**Comment by agjohnson** [comment]
Confirmed that urllib3 *will not* add a deprecation exception like urllib3 2.0 did. Because of this we will not have to worry about builds failing due to outdated libssl come Sept 11.

So, we can now set a date later in Sept (or even call it the first week of Oct) to give a bit more room for this migration timeframe.

We also don't need to be strict about the deprecation date now either. If projects aren't migrating over, we had talked about allowing some projects to continue building after the cutoff (big or long standing projects having trouble with the migration).

**Comment by ericholscher** [comment]
@agjohnson I guess I don't understand the reasoning for changing the dates we agreed to on the call? Is there a specific worry you're trying to address?

**Comment by agjohnson** [comment]
Just gaining 3 weeks on our timeframe. We described the timeframe as fairly tight. We could rather quickly decide on Monday Sept 25 and keep everything else the same.

**Comment by ericholscher** [comment]
I'm fine with that date, and gives us room for a 3rd brownout if we need it.

**Comment by agjohnson** [comment]
That's a great idea. We could add another 24h or even 48h brownout on Sept 4 in place of the cutover.

Looks like everyone is planning on being around Mon Sept 25 too, just in case. We will have been through this 3 times at that point though, so I doubt there will be many fires then.

File: migrate-configuration-v2.rst
Comment (agjohnson) [inline migrate-configuration-v2.rst:18]:
```suggestion
**We plan to start failing builds not using configuration file version 2 on September 25, 2023**.
```

File: migrate-configuration-v2.rst
Comment (agjohnson) [inline migrate-configuration-v2.rst:39]: Adding a third brownout, evenly space brownouts at 3 weeks apart (Aug 7 brownout is Aug 14 brownout). Team is available on all these dates, based on our shared calendar.

```suggestion
* **Monday, July 24, 2023**: Do the first brownout (temporarily enforce this deprecation) for 12 hours: 00:01 PST to 11:59 PST (noon)
* **Monday, August 14, 2023**: Do a second brownout (temporarily enforce this deprecation) for 24 hours: 00:01 PST to 23:59 PST (midnight)
* **Monday, September 4, 2023**: Do a third and final brownout (temporarily enforce this deprecation) for 48 hours: 00:01 PST to 23:59 PST (midnight)
* **Monday, September 25, 2023**: Fully remove support for building documentation without configuration file v2.
```

**Review by agjohnson — APPROVED** [review]
Two changes:

- Bump brownout Aug 7 -> Aug 14 to evenly space
- Sep 4 brownout for 48h added

If 24h seems more right, lets go with that though.

**Review by humitos — APPROVED** [review]
This looks good to me with Anthony's suggestions 👍

**Pull request merged**
